### ISSUE TITLE
Define confidence intervals for verifications when frame random sampling is applied

### ISSUE THREAD

**Issue by ndujar** [livepeer/verification-classifier#67]
When making verifications, the inference process can be optimized by taking at random a reduced number of frames from the segment (frame random sampling).
Although it doesn't affect directly to the accuracy of the classifier, introduces en element of uncertainty that needs to be accounted for.
In order to extract those confidence intervals, the verifier needs to be tested under different values of Frame Random Sampling (FRS) in successive iterations. The methodology to follow is described [here](http://people.stern.nyu.edu/churvich/Regress/Handouts/Chapt2.pdf), and we will apply it by making a large sample of TNR/TPR values for each FRS in the list [5, 10, 15, 20, 25, 30, 35, 40 ,45, 50, 55 ,60].

**Comment by ndujar** [comment]
A series of experiments has been done in order to determine to what extent the choice of a number of samples affects accuracy.
We have trained a OCSVM model with the full sequence of frames (up to 60 when available) using the well encoded assets (\~15k) of the dataset. This is the same procedure employed for the creation of the model used in the API (same features, same number of training specimens).
In order to test its validity, we have measured its accuracy in terms of True Positive Rate, True Negative Rate and F-score. The number of tampered videos was \~180k, and the remaining \~4k we legit assets used for testing. Results are shown in table below:

|#samples|f20|tnr|tpr_train|tpr_test|
|----|----|----|----|----|
|60|0.967821|0.886745|0.986831|0.980774|

Using this trained model, we have iteratively repeated the prediction passing to it the average of the values of frames at random positions for each and every asset. We have chosen a repetition number of 500 to have a large enough population. The number of frames has been increased in steps of 5, between 5 and 55.
Histograms below show the different distributions for all eleven groups of experiments:

![image.png](https://images.zenhubusercontent.com/5cf0489817221c71b75cc4e4/751b9b5e-abb7-4709-8e57-7cf383178241)

In order to assess the goodness-of-fit measure of departure from normality, Shappiro-Wilk and D'Agostino K-squared tests have carried away over the above depicted data. Results of the p-values are display in the table below:

|# Frames|Shapiro|DAgostino|
|----|----|----|
|5|0.000099|0.000142|
|10|0.058814|0.058678|
|15|0.778593|0.760594|
|20|0.844955|0.863694|
|25|0.477455|0.239329|
|30|0.902962|0.690180|
|35|0.498511|0.476472|
|40|0.695625|0.298841|
|45|0.269293|0.370202|
|50|0.768750|0.533112|
|55|0.654181|0.349123|

Using 5% as the limit of the p-value (alpha-level < 0.05), we can assume that, other than the 5 frame set of experiments, the null hypothesis of  the distributions being normal is satisfied.

With this assumption, we have elaborated the following tables agggregating the mean values after all experiments and their respective deltas:

**Mean accuracy values after 500 experiments**

**#S f20 **|**tnr **|**tpr\_train **|**tpr\_test**
:-----:|:-----:|:-----:|:-----:
5 |0.954069 |0.891472 |0.976687 
10 |0.961570 |0.889883 |0.982999 
15 |0.963884 |0.888872 |0.985204 
20 |0.965269 |0.888295 |0.985927 
25 |0.965990 |0.887916 |0.986478 
30 |0.966495 |0.887667 |0.986973 
35 |0.966790 |0.887444 |0.987309 
40 |0.967037 |0.887321 |0.987474 
45 |0.967293 |0.887176 |0.987707 
50 |0.967448 |0.887108 |0.987751 
55 |0.967602 |0.886989 |0.987932 

**Confidence deltas 99%**

 **#Frames**|**f20 **|**tnr **|**tpr\_train **|**tpr\_test**
:-----:|:-----:|:-----:|:-----:|:-----:
5 |0.000260 |0.000189 |0.000123 |0.000265
10 |0.000212 |0.000125 |0.000103 |0.000217
15 |0.000169 |0.000097 |0.000094 |0.000172
20 |0.000167 |0.000086 |0.000088 |0.000169
25 |0.000143 |0.000074 |0.000093 |0.000146
30 |0.000137 |0.000066 |0.000091 |0.000140
35 |0.000130 |0.000066 |0.000087 |0.000132
40 |0.000123 |0.000061 |0.000088 |0.000125
45 |0.000114 |0.000056 |0.000084 |0.000116
50 |0.000113 |0.000051 |0.000086 |0.000115
55 |0.000106 |0.000048 |0.000087 |0.000107

In summary, what these results tell us is that, for 99% of the experiments, the achieved accuracy will fall very close (within a \~1000% up or down) of the mean values obtained with 60 frames, regardless of the number of frames chosen.
